modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Keywords.js

```diff
@@ -38,122 +39,197 @@ export default function Keywords({currentPage, languageTag}) {
 
 	const [{publishedToday}] = useContext(StoreContext);
 
+	const keywordsData = useMemo(() => {
+		const dataKeys = new Set();
+
+		const countries = currentPage.data.countryKeywords.reduce(
+			(acc, {countryCode, countryName}) => {
+				if (dataKeys.has(countryCode)) {
+					return acc;
+				}
+
+				dataKeys.add(countryCode);
+
+				return acc.concat({countryCode, countryName});
+			},
+			[]
+		);
+
+		return {countries};
+	}, [currentPage.data.countryKeywords]);
+
+	const {countries} = keywordsData;
```

Comment:
I'll go with:

```js
const keywords = useMemo(() => {
	const countryKeywords =
		currentCountry &&
		currentPage.data.countryKeywords.find((country) => {
			return country.countryCode === currentCountry;
		});

	return countryKeywords?.keywords ?? [];
}, [currentPage.data.countryKeywords, currentCountry]);

const handleCountrySelection = (event) => {
	const country = event.target.value;
	setCurrentCountry(country);
};
```

Because I'm only retrieving countries :) Thanks!